src/xSTIR/cSTIR/cstir_p.cpp

```diff
@@ -395,18 +395,18 @@ sirf::cSTIR_PLSPriorParameter
 	else if (boost::iequals(name, "eta"))
 		return dataHandle<float>(prior.get_eta());
 	else if (boost::iequals(name, "anatomical_image")) {
-		sptrImage3DF sptr_im = prior.get_anatomical_image_sptr();
-		shared_ptr<STIRImageData> sptr_id(new STIRImageData(sptr_im));
+		auto sptr_im = prior.get_anatomical_image_sptr();
+		shared_ptr<STIRImageData> sptr_id(new STIRImageData(*sptr_im->clone()));
```

Comment:
```suggestion
		shared_ptr<STIRImageData> sptr_id(sptr_im->clone());
```
This avoids one copy. Same for lines below

Reply:
Actually, I think the original line would leak the cloned image. The new one won't.

Reply:
Hm that doesn't work (remember that sptr_im is type `shared_ptr<const DiscDensity>`, not another `STIRImageData`).

Reply:
hmmm. The `shared_ptr` constructor should forward to the actual object constructor. Maybe that one's `explicit`?

Reply:
it isn't. we just have
```
STIRImageData(const Image3DF& image)
```
so now I'm confused why this wouldn't compile

Reply:
Image3DF is VoxelsOnCartesian, whereas sptr_im is a DiscDensity?

Reply:
http://www.cplusplus.com/reference/memory/shared_ptr/shared_ptr/ lists for instance
```c++
template <class U> explicit shared_ptr (U* p);
```
So, this should work I think. Did you actually try?

Reply:
Yeah I tried.

Reply:
`typedef stir::DiscretisedDensity<3, float> Image3DF;`

Reply:
what was the error message?

Reply:
Something long and confusing. Could it be because it's a `stir::shared_ptr`, not a `std::shared_ptr`?

Error: 

```
/Users/rich/Documents/Code/SIRF-SuperBuild/Build/sources/SIRF/src/xSTIR/cSTIR/cstir_p.cpp:399:29: error: no matching constructor for initialization of 'shared_ptr<sirf::STIRImageData>'
                shared_ptr<STIRImageData> sptr_id(sptr_im->clone());
                                          ^       ~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:3756:23: note: candidate constructor not viable: no known conversion from 'DiscretisedDensity<3, float> *' to 'std::nullptr_t' (aka 'nullptr_t') for 1st argument
    _LIBCPP_CONSTEXPR shared_ptr(nullptr_t) _NOEXCEPT;
                      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:3770:5: note: candidate constructor not viable: no known conversion from 'DiscretisedDensity<3, float> *' to 'const std::__1::shared_ptr<sirf::STIRImageData>' for 1st argument
    shared_ptr(const shared_ptr& __r) _NOEXCEPT;
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:3778:5: note: candidate constructor not viable: no known conversion from 'DiscretisedDensity<3, float> *' to 'std::__1::shared_ptr<sirf::STIRImageData>' for 1st argument
    shared_ptr(shared_ptr&& __r) _NOEXCEPT;
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:3758:18: note: candidate template ignored: requirement 'is_convertible<stir::DiscretisedDensity<3, float> *, sirf::STIRImageData *>::value' was not satisfied [with _Yp = stir::DiscretisedDensity<3, float>]
        explicit shared_ptr(_Yp* __p,
```

Reply:
`stir::shared_ptr` should be `std::shared_ptr` as we're using C++11.

I'm baffled.

what about
```
shared_ptr<Image3DF> stir_sptr(sptr_im->clone());
shared_ptr<STIRImageData> sptr_id(stir_sptr);
```
although that seems even shakier to me

Reply:
No luck either unfortunately.

Reply:
Then we have not much choice than making 2 copies, but at least not leaking
```c++
shared_ptr<Image3DF> stir_sptr(sptr_im->clone());
shared_ptr<STIRImageData> sptr_id = std::make_shared<STIRImageData>(*stir_sptr);
```
unless we'd create constructors for `STIRImageData` that accept a `shared_ptr`.

Reply:
I went for the following:
```
auto sptr_im = prior.get_anatomical_grad_sptr(dir);
auto sptr_id = std::make_shared<STIRImageData>(*sptr_im);
```

Reply:
I suppose in the original commit the `clone()` was never needed, since the `STIRImageData` constructor takes a `const DiscDensity &` as argument. Not sure why I put it in.

Reply:
excellent. thanks!